ui/src/app/pages/workspace/workspace-about.tsx

```diff
@@ -178,20 +185,25 @@ export const WorkspaceAbout = fp.flow(
       this.loadUserRoles(workspace);
     }
 
-    async loadInitialCreditsUsage(workspace: WorkspaceData) {
-      const usage = await workspacesApi().getBillingUsage(
-        workspace.namespace,
-        workspace.id
+    loadInitialCreditsUsage(workspace: WorkspaceData) {
+      fetchWithErrorModal(
+        () =>
+          workspacesApi().getBillingUsage(workspace.namespace, workspace.id),
+        {
+          onSuccess: (usage: WorkspaceBillingUsageResponse) =>
+            this.setState({ workspaceInitialCreditsUsage: usage.cost }),
+        }
```

Comment:
I think you want to avoid passing a callback. This is the older asynchronous paradigm and the chaining style (via promises) allows us to avoid nesting. Maybe you can try to target something like this:
```suggestion
      fetchWithErrorModal(
        () => workspacesApi().getBillingUsage(workspace.namespace, workspace.id)
      ).then((usage: WorkspaceBillingUsageResponse) =>
        this.setState({ workspaceInitialCreditsUsage: usage.cost }))
```

Reply:
callback removed